src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -18008,3 +18011,53 @@ def timedelta_property(
         return SnowflakeQueryCompiler(
             self._modin_frame.apply_snowpark_function_to_columns(func, include_index)
         )
+
+    def create_qc_with_extra_columns(
+        self, extra_columns: List[Hashable]
+    ) -> "SnowflakeQueryCompiler":
+        """
+        This is a helper function for creating a DataFrame where the data is a DataFrame object. Sometimes, columns
+        not present in the `data` DataFrame can be passed as arguments - these are added to the resultant DataFrame
+        as NaN columns.
+
+        Parameters
+        ----------
+        extra_columns : list of hashable, default None
+            New columns that are not a part of the original query compiler
+
+        Returns
+        -------
+        SnowflakeQueryCompiler
+            A new query compiler with the new columns.
+        """
+        self_frame = self._modin_frame
+
+        if not extra_columns or len(extra_columns) == 0:
+            return self.copy()
+
+        # Append the new columns to the data's internal frame.
+        new_snowflake_quoted_identifiers = (
+            self._modin_frame.ordered_dataframe.generate_snowflake_quoted_identifiers(
+                pandas_labels=extra_columns,
+                excluded=self_frame.data_column_snowflake_quoted_identifiers,
+            )
+        )
+        new_ordered_frame = append_columns(
+            self_frame.ordered_dataframe,
+            new_snowflake_quoted_identifiers,
+            [pandas_lit(np.nan)] * len(extra_columns),
```

Comment:
what's the type of the new column? is there a difference between np.nan/None here?

Reply:
In pandas you can create DataFrames and specify columns that don't exist as values to be part of the final DataFrame. This case arises when you supply a DataFrame `df` as the data, and the columns supplied include columns not present in `df`. The extra columns are created as NaN columns.

```py
>>> pd.DataFrame(pd.DataFrame({"A": [1, 2], "B": [3, 4]}), columns=["A", "C"])
   A   C
0  1 NaN
1  2 NaN
```